### PR TITLE
homework_4

### PR DESCRIPTION
**Pull request by cadmiumyellow**

### CODE REVIEW

File: src/test/java/ru/training/at/hw4/listener/ScreenshotListener.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    @Attachment(value = "Page screenshot", type = "image/png")
+    public byte[] saveScreenshotPng(WebDriver driver) {
+        return ((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES);
+    }
+
+    @Override
+    public void onTestFailure(ITestResult testResult) {
+
+        Object testClass = testResult.getInstance();
+        WebDriver driver = ((BaseTest) testClass).webDriver;
```
Comment (DmitryKhodakovsky) [inline src/test/java/ru/training/at/hw4/listener/ScreenshotListener.java]: Why do you decide do not use `TestContext` from TestNg?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please resolve comments and make your allure report more informative as we discussed

File: src/test/java/ru/training/at/hw4/listener/ScreenshotListener.java (hunk 1 above)
Comment (cadmiumyellow) [reply src/test/java/ru/training/at/hw4/listener/ScreenshotListener.java]: Ok, used TestContext instead

**Review by DmitryKhodakovsky — APPROVED** [review]
HW4 approved

**Comment by DmitryKhodakovsky** [comment]
HW4 accepted, please resolve conflicts and merge it to the master branch

**Pull request merged**